packages/@aws-cdk/aws-sns/lib/subscription.ts

```diff
@@ -115,27 +125,51 @@ export class Subscription extends Resource {
       if (total > 100) {
         throw new Error(`The total combination of values (${total}) must not exceed 100.`);
       }
+    } else if (props.filterPolicyWithMessageBody) {
+      if (Object.keys(props.filterPolicyWithMessageBody).length > 5) {
```

Comment:
```suggestion
      // https://docs.aws.amazon.com/sns/latest/dg/subscription-filter-policy-constraints.html
      if (Object.keys(props.filterPolicyWithMessageBody).length > 5) {
```

For our future selves.